### PR TITLE
Update oc installation to include instructions by OS

### PR DESCRIPTION
**Pull request by bergerhoffer**
Preview:

* https://pr-21346-updates--ocpdocs.netlify.app/openshift-enterprise/latest/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands (OCP)
* https://pr-21346-updates--ocpdocs.netlify.app/openshift-origin/latest/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands (OKD)

Replaces #21346.

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@jankleinert Here is the replacement PR with some updates. Mostly minor except for the Windows section, where I removed the specifics on setting the PATH, which I think is outside the scope of what OCP docs should be providing steps for. Let me know if you have any feedback. Thanks!

Preview: https://pr-21346-updates--ocpdocs.netlify.app/openshift-enterprise/latest/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands

File: modules/cli-installing-cli.adoc [hunk 1]
```diff
@@ -28,8 +28,8 @@
 [id="cli-installing-cli_{context}"]
 = Installing the CLI by downloading the binary
 
-You can install the CLI in order to interact with {product-title} using a
-command-line interface.
+You can install the OpenShift CLI (`oc`) in order to interact with {product-title} using a
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:31]: s/using/by using/

File: modules/cli-installing-cli.adoc [hunk 2]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
 install the new version of `oc`.
 ====
 
+[id="cli-installing-cli-on-linux_{context}"]
+== Installing the CLI on Linux
+
+You can install the OpenShift CLI (`oc`) binary on Linux using the following procedure.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:44]: s/using/by using/

File: modules/cli-installing-cli.adoc [hunk 3]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
-. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system
-. Download `oc.tar.gz`
+. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system and architecture.
+. Download `oc.tar.gz`.
 endif::[]
 ifndef::openshift-origin[]
-. From the link:https://cloud.redhat.com/openshift/install[Infrastructure Provider] page on the {cloud-redhat-com} site, navigate to the page for your installation type and
-click *Download Command-line Tools*.
+. From the link:https://cloud.redhat.com/openshift/install[Infrastructure Provider] page on the {cloud-redhat-com} site, navigate to the page for your installation type.
+. In the *Command-line interface* section, select *Linux* from the drop-down menu and click *Download command-line tools*.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:54]: s/menu and click/menu, and then click/

File: modules/cli-installing-cli.adoc [hunk 4]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
 ifdef::openshift-origin[]
-. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system
-. Download `oc.tar.gz`
+. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system and architecture.
+. Download `oc.tar.gz`.
 endif::[]
 ifndef::openshift-origin[]
-. From the link:https://cloud.redhat.com/openshift/install[Infrastructure Provider] page on the {cloud-redhat-com} site, navigate to the page for your installation type and
-click *Download Command-line Tools*.
+. From the link:https://cloud.redhat.com/openshift/install[Infrastructure Provider] page on the {cloud-redhat-com} site, navigate to the page for your installation type.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:53]: Do we want to be more specific? First a user selects the infrastructure page, and then the installation type page.

File: modules/cli-installing-cli.adoc [hunk 5]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
-====
-You can install `oc` on Linux, Windows, or macOS.
-====
-. Save the file to your file system.
-. Extract the compressed file.
-. Place it in a directory that is on your `PATH`.
+----
+$ tar xvzf <file>
+----
+. Place the `oc` binary in a directory that is on your `PATH`.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:61]: s/on your/in your/

File: modules/cli-installing-cli.adoc [hunk 6]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+After you install the CLI, it is available using the `oc` command:
+
+----
+$ oc <command>
+----
+
+[id="cli-installing-cli-on-windows_{context}"]
+== Installing the CLI on Windows
+
+You can install the OpenShift CLI (`oc`) binary on Windows using the following procedure.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:78]: s/using/by using/

File: modules/cli-installing-cli.adoc [hunk 7]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+
+.Procedure
+
+ifdef::openshift-origin[]
+. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system and architecture.
+. Download `oc.zip`.
+endif::[]
+ifndef::openshift-origin[]
+. From the link:https://cloud.redhat.com/openshift/install[Infrastructure Provider] page on the {cloud-redhat-com} site, navigate to the page for your installation type.
+. In the *Command-line interface* section, select *Windows* from the drop-down menu and click *Download command-line tools*.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:88]: Some question about including additional steps/specificity for the procedure.

File: modules/cli-installing-cli.adoc [hunk 8]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+endif::[]
+. Unzip the archive with a ZIP program.
+. Move the `oc` binary to a directory that is on your `PATH`.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:91]: s/on your/in your/

File: modules/cli-installing-cli.adoc [hunk 9]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+After you install the CLI, it is available using the `oc` command:
+
+----
+C:\> oc <command>
+----
+
+[id="cli-installing-cli-on-macos_{context}"]
+== Installing the CLI on macOS
+
+You can install the OpenShift CLI (`oc`) binary on macOS using the following procedure.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:108]: s/using/by using/

File: modules/cli-installing-cli.adoc [hunk 10]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+== Installing the CLI on Linux
+
+You can install the OpenShift CLI (`oc`) binary on Linux using the following procedure.
+
 .Procedure
+
 ifdef::openshift-origin[]
-. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system
-. Download `oc.tar.gz`
+. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system and architecture.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:49]: s/Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the/Navigate to the https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ page, and then choose the/

File: modules/cli-installing-cli.adoc [hunk 11]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+
+[id="cli-installing-cli-on-windows_{context}"]
+== Installing the CLI on Windows
+
+You can install the OpenShift CLI (`oc`) binary on Windows using the following procedure.
+
+.Procedure
+
+ifdef::openshift-origin[]
+. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system and architecture.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:83]: s/Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the/Navigate to the https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ page, and then choose the/

File: modules/cli-installing-cli.adoc [hunk 12]
```diff
@@ -38,25 +38,93 @@ of the commands in {product-title} {product-version}. Download and
...
+
+[id="cli-installing-cli-on-macos_{context}"]
+== Installing the CLI on macOS
+
+You can install the OpenShift CLI (`oc`) binary on macOS using the following procedure.
+
+.Procedure
+
+ifdef::openshift-origin[]
+. Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the folder for your operating system and architecture.
```
Comment (jboxman) [inline modules/cli-installing-cli.adoc:113]: s/Navigate to https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ and choose the/Navigate to the https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/ page, and then choose the/

**Review by jboxman — APPROVED** [review]
@bergerhoffer, done. Minor picks.

File: modules/cli-installing-cli.adoc (hunk 1 above)
Comment (bergerhoffer) [reply modules/cli-installing-cli.adoc:31]: Changed to "from".

File: modules/cli-installing-cli.adoc (hunk 5 above)
Comment (bergerhoffer) [reply modules/cli-installing-cli.adoc:61]: Leaving, as "on" sounds right to me, and we have a mixed bag in the doc set now, with the majority going to "on".

File: modules/cli-installing-cli.adoc (hunk 3 above)
Comment (bergerhoffer) [reply modules/cli-installing-cli.adoc:54]: I believe that either is fine. I checked the IBM style guide and see examples such as "In the Password field, type the password and click OK."

**Comment by bergerhoffer** [comment]
Hi @zhouying7780, could you please review this update? Thanks!

Previews:
* https://pr-21346-updates--ocpdocs.netlify.app/openshift-enterprise/latest/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands (OCP)
* https://pr-21346-updates--ocpdocs.netlify.app/openshift-origin/latest/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands (OKD)

**Comment by jankleinert** [comment]
> @jankleinert Here is the replacement PR with some updates. Mostly minor except for the Windows section, where I removed the specifics on setting the PATH, which I think is outside the scope of what OCP docs should be providing steps for. Let me know if you have any feedback. Thanks!

 @bergerhoffer This works and looks clear. LGTM, thanks!

**Comment by bergerhoffer** [comment]
Cherry picks to 4.5 (and earlier) are still needed. Waiting to decide whether previous commits (that caused the conflict) need to be cherry picked to 4.5 (and potentially earlier) first.

**Pull request merged**